src/sing/phraseRendering.ts

```diff
@@ -0,0 +1,803 @@
+import {
```

Comment:
このファイルとフレーズレンダラーが何をするものなのか、初見だとかなり見えて来づらいので、そこそこ詳しめに説明があるとわかりやすいと思います！
とりあえずここは･･･これくらいとか？


```suggestion
/**
 * フレーズごとに音声合成するフレーズレンダラーと、それに必要な処理。
 * キャッシュの作成や判定も行う。
 */

import {
```

（キャッシュの話はStageのとこでも良いかも）

Reply:
ファイルの先頭にコメントを追加しました！